dsp_permissions_scripts/oap/oap_get.py

```diff
@@ -30,6 +33,82 @@
     "knora-api:userHasPermission",
     "knora-api:hasPermissions",
 ]
+KB_RESCLASSES = [
+    "knora-api:VideoSegment",
+    "knora-api:AudioSegment",
+    "knora-api:Region",
+    "knora-api:Annotation",
+    "knora-api:LinkObj",
+]
+
+
+def get_oaps_of_kb_resclasses(dsp_client: DspClient, project_iri: str, oap_config: OapRetrieveConfig) -> list[Oap]:
+    match oap_config.retrieve_resources:
+        case "specified_res_classes":
+            specified_kb_resclasses = [x for x in KB_RESCLASSES if x in oap_config.specified_res_classes]
+            res_only_oaps = _get_oaps_of_specified_kb_resclasses(dsp_client, project_iri, specified_kb_resclasses)
```

Comment:
don't you get a reference before assignment problem here because there is no `case _:`

Reply:
The type of `oap_config.retrieve_resources` is `Literal['all', 'specified_res_classes']`, so it wouldn't make sense to have a `case _`